share/completions/git.fish

```diff
@@ -737,16 +661,39 @@ function __fish_git_complete_stashes
     __fish_git stash list --format=%gd:%gs 2>/dev/null | string replace ":" \t
 end
 
-function __fish_git_aliases
-    __fish_git config -z --get-regexp '^alias\.' 2>/dev/null | while read -lz key value
-        begin
-            set -l name (string replace -r '^.*\.' '' -- $key)
-            set -l val (string shorten --no-newline -m 36 -- $value)
-            printf "%s\t%s\n" $name "alias: $val"
+function __fish_git_get_aliases
+    # Get valid aliases - aliases for a builtin or custom commands are ignored
+    set -l non_aliases (__fish_git --list-cmds=main,others)
```

Comment:
From https://git-scm.com/docs/git:

> This is an internal/experimental option and may change or be removed in the future.

What it does is complete "main (all commands in libexec directory)" and "others (all other commands in $PATH that have git- prefix)". That's *okay* to reconstruct - "others" is really just `$PATH/git-*`, main would require us to check git's libexec dir, I don't know how to get that portably.